### PR TITLE
Support multi-line code blocks in examples

### PR DESCRIPTION
**Pull request by Greg-Smulko**
# PR Summary

Support multi-line code blocks in examples.

## PR Context

Fix as asked for by @vors in https://github.com/PowerShell/platyPS/issues/180#issuecomment-540895324. 
Fixes the issue https://github.com/PowerShell/platyPS/issues/180 _Multiline examples don't display as markdown code_.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
@Greg-Smulko Please rebase to pass Static CI.

File: src/System.Management.Automation/help/HelpCommentsParser.cs [hunk 1]
```diff
@@ -493,49 +493,27 @@ private void BuildSyntaxForParameterSet(XmlElement command, XmlElement syntax, M
             command.AppendChild(syntax).AppendChild(syntaxItem);
         }
 
-        private static void GetExampleSections(string content, out string prompt_str, out string code_str, out string remarks_str)
+        internal static void GetExampleSections(string content, out string prompt_str, out string code_str, out string remarks_str)
```
Comment (iSazonov) [inline src/System.Management.Automation/help/HelpCommentsParser.cs]: Can we use a public API for a test? Or `internal XmlDocument BuildXmlFromComments()`?

Comment (Greg-Smulko) [reply src/System.Management.Automation/help/HelpCommentsParser.cs]: Yes, I thought about it too.

Then I looked around and decided that the setup and assertions would be much more complicated, so overall it would not be worth it really IMO.

Comment (iSazonov) [reply src/System.Management.Automation/help/HelpCommentsParser.cs]: Perhaps we could find a script repo and make Pester test.

Comment (Greg-Smulko) [reply src/System.Management.Automation/help/HelpCommentsParser.cs]: I looked around again and found https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Language/Scripting/ScriptHelp.Tests.ps1

It's weird that I didn't spot it earlier...

Extending these PS tests should work fine. I can give it a try, but probably not before next weekend.

**Comment by iSazonov** [comment]
Add WIP until we get Pester tests.

**Comment by Greg-Smulko** [comment]
I replaced unit tests with Pester tests, could you take a look again @iSazonov?

**Comment by iSazonov** [comment]
@Greg-Smulko Could you please check the fix with PlatyPS?

File: test/powershell/Language/Scripting/ScriptHelp.Tests.ps1 [hunk 2]
```diff
@@ -627,4 +627,64 @@ Describe 'get-help other tests' -Tags "CI" {
...
+        It '$x.examples.example[3].introduction[0].text' { $x.examples.example[3].introduction[0].text | Should -BeExactly "PS>" }
+        It '$x.examples.example[3].code' { $x.examples.example[3].code | Should -BeExactly "Get-Service`nOutput" }
+        It '$x.examples.example[3].remarks[0].text' { $x.examples.example[3].remarks[0].text | Should -BeExactly "Explanation.`nSecond line." }
+        It '$x.examples.example[3].remarks.length' { $x.examples.example[3].remarks.length | Should -Be 5 }
+        It '$x.examples.example[4].introduction[0].text' { $x.examples.example[4].introduction[0].text | Should -BeExactly "PS>" }
+        It '$x.examples.example[4].code' { $x.examples.example[4].code | Should -BeExactly "Get-Service`nOutput" }
+        It '$x.examples.example[4].remarks[0].text' { $x.examples.example[4].remarks[0].text | Should -BeExactly "Explanation.`n`nNext section." }
+        It '$x.examples.example[4].remarks.length' { $x.examples.example[4].remarks.length | Should -Be 5 }
+    }
+
```
Comment (iSazonov) [inline test/powershell/Language/Scripting/ScriptHelp.Tests.ps1]: Please remove the extra line.
```suggestion
```

**Comment by Greg-Smulko** [comment]
> Could you please check the fix with PlatyPS?

@iSazonov , I tested the fix with PlatyPS, everything looks fine. :)
See the [files attached](https://github.com/PowerShell/PowerShell/files/3744946/test-examples-generation.zip).
I generated them using the `New-MarkdownHelp` and `New-ExternalHelp` cmdlets.

**Review by iSazonov — APPROVED** [review]
@Greg-Smulko Thanks for confirmation!

I think it would be useful to measure and share a performance degradation since we use Regex now. Could you please do this?

**Comment by Greg-Smulko** [comment]
@iSazonov , performance degradation? What degradation?
On `master` the `Measure-Command -Expression { Get-Help Get-Help -examples }` took 304 ms, on the new version only 287 ms.

But seriously, it seems that the very first run of `Get-Help` takes around 300 ms on both versions, and then the subsequent ones (for a different command) about 10-15 ms.

I tried running `Measure-Command -Expression { Get-Help Get-Help -examples }` (took ~300 ms) and then `Measure-Command -Expression { Get-Help Get-Command -examples }` (took ~15 ms).

To summarize, I think that there is no visible performance degradation at all, at least for the cmdlets that I tested.

**Comment by iSazonov** [comment]
@Greg-Smulko Thanks!

@adityapatwardhan Please review the PR.

**Comment by Greg-Smulko** [comment]
Hi @adityapatwardhan, any hope that you could take a look at this?

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@Greg-Smulko  Thank you for your contribution!
